datafusion/src/datasource/mod.rs

```diff
@@ -36,3 +43,247 @@ pub(crate) enum Source<R = Box<dyn std::io::Read + Send + Sync + 'static>> {
     /// Read data from a reader
     Reader(std::sync::Mutex<Option<R>>),
 }
+
+#[derive(Debug, Clone)]
+/// A single file that should be read, along with its schema, statistics
+/// and partition column values that need to be appended to each row.
+pub struct PartitionedFile {
+    /// Path for the file (e.g. URL, filesystem path, etc)
+    pub file_path: String,
+    /// Statistics of the file
+    pub statistics: Statistics,
```

Comment:
I think this makes sense to keep as `Statistics` rather than, say `Option<Statistics>` all the fields are already `Option`